src/odemis/gui/cont/stream.py

```diff
@@ -129,18 +129,35 @@ def __init__(self, stream_bar, stream, tab_data_model, show_panel=True, view=Non
         if hasattr(stream, "power"):
             va = self.stream.power
             name = "power"
+            hw_settings = self.tab_data_model.main.hw_settings_config
+
+            # if there is a light component connected to the stream to be able to control the power
+            # set the emitter configuration to this component instead of the emitter (usually the e-beam)
+            if hasattr(self.stream, "light"):
+                light_config = get_hw_config(self.stream.light, hw_settings)
+            else:
+                light_config = {}
+
+            l_conf = light_config.get(name)
+            if l_conf is not None:
+                logging.debug("%s light configuration found for %s", name,
+                              self.stream.light.role)
+
             if self.stream.emitter:
-                hw_settings = self.tab_data_model.main.hw_settings_config
-                emitter_conf = get_hw_config(self.stream.emitter, hw_settings)
+                emitter_config = get_hw_config(self.stream.emitter, hw_settings)
             else:
-                emitter_conf = {}
+                emitter_config = {}
 
-            conf = emitter_conf.get(name)
-            if conf is not None:
+            e_conf = emitter_config.get(name)
+            if e_conf is not None:
                 logging.debug("%s emitter configuration found for %s", name,
                               self.stream.emitter.role)
 
-            self.add_setting_entry(name, va, self.stream.emitter, conf)
+            if self.stream.light:
+                self.add_setting_entry(name, va, self.stream.light, l_conf)
+
+            if self.stream.emitter:
+                self.add_setting_entry(name, va, self.stream.emitter, e_conf)
```

Comment:
Sorry, I had misunderstood the context of the code in my initial comment. It should be added only once at most, as there is just one "power" VA.
I think that this can be done very simply compared to the original code. Something like that:
```
        if hasattr(stream, "power"):
            va = self.stream.power
            name = "power"
            if hasattr(self.stream, "light"):
                comp = self.stream.light
            else:
                comp = self.stream.emitter
            
            if comp is not None:
                hw_settings = self.tab_data_model.main.hw_settings_config
                emitter_conf = get_hw_config(comp, hw_settings)
            else:
                emitter_conf = {}

            conf = emitter_conf.get(name)
            if conf is not None:
                logging.debug("%s emitter configuration found for %s", name,
                              comp.role)

            self.add_setting_entry(name, va, comp, conf)
```